src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -864,6 +864,23 @@ private static bool PackBlocks(ArrayBuilder<BasicBlockBuilder> blocks, PooledDic
                                         continue;
                                     }
 
+                                    if (currentRegion.LastBlock == block &&
+                                        block.BranchValue is IFlowCaptureReferenceOperation { Id: var id } &&
```

Comment:
>block.BranchValue is IFlowCaptureReferenceOperation { Id: var id } [](http://example.com/codeflow?start=40&length=66)

As we discussed offline, we probably should be able to handle more complicated experssions using ```IFlowCaptureReferenceOperation``` inside.